wlroots/wlr_types/texture.py

```diff
@@ -34,3 +35,33 @@ def from_pixels(
             renderer._ptr, fmt, stride, width, height, data
         )
         return Texture(ptr)
+
+    def write_pixels(
+        self,
+        stride: int,
+        width: int,
+        height: int,
+        data: ffi.CData,
+        src_x: int = 0,
+        src_y: int = 0,
+        dst_x: int = 0,
+        dst_y: int = 0,
+    ) -> bool:
+        """
+        Update a texture with raw pixels. The texture must be mutable, and the input
+        data must have the same pixel format that the texture was created with.
+
+        Should not be called in a rendering block like renderer_begin()/end() or
+        between attaching a renderer to an output and committing it.
+
+        data must be a CData pointer to pixel data.
+        """
+        return lib.wlr_texture_write_pixels(
+            self._ptr, stride, width, height, src_x, src_y, dst_x, dst_y, data,
+        )
+
+    def destroy(self) -> None:
+        """Destroys this wlr_texture."""
+        if self._ptr is not None:
+            lib.wlr_texture_destroy(self._ptr)
```

Comment:
Should this be set as the `gc` action when creating from `wlr_texture_from_pixels`?

Reply:
Yep sounds good. Updated to reflect that.

Reply:
Actually, hold on this, I think this breaks things

Reply:
The gc approach doesn't work well because textures are not always created and held/owned in Python for their entire lifetime, sometimes they are held in `Texture` temporarily and should be allowed to persist after the `Texture` is gone, at the C wlroots level. Having the method to destroy textures is useful to destroy them manually, but this is only useful for textures created manually (i.e. by the compositor, what I'm doing with the `Bar`, for example), while other textures created by wlroots will be destroyed by wlroots (though a `Texture` will exist at some point for a given texture, transiently).

So I've reverted this back to the `destroy` method.